### PR TITLE
test: add code coverage

### PR DESCRIPTION
**Pull request by jayachithra**
1. Added extension to display coverage reports
2. Updated docs with steps on how to generate and visualize code coverage

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -33,6 +33,24 @@ To run linting, run the following command:
...
+To run all tests and generate coverage data run the following command:
+
+```bash
+poetry run coverage run --source=src -m pytest .
+```
+
+This will generate a .coverage file. To visualize code coverage, generate a coverage.xml file from the .coverage file by running:
+
+```bash
+poetry run coverage xml
```
Comment (arjendev) [inline DEVELOPMENT.md:49]: Awesome, works immediately as expected :) 

Super super super nit: we could chain the commands: `poetry run coverage run --source=src -m pytest . && coverage xml`

**Review by arjendev — DISMISSED** [review]
Awesome addition! 

For the future, as we talked about, we can consider checking to automatically run the coverage as part of running tests and we could also look into generating a report with the pipeline so we can view the coverage as part of PR review.

**Comment by jayachithra** [comment]
> Awesome addition!
> 
> For the future, as we talked about, we can consider checking to automatically run the coverage as part of running tests and we could also look into generating a report with the pipeline so we can view the coverage as part of PR review.

Good idea to integrate coverage into VSCode test explorer, it's probably worth considering pytest-cov [python 3.x - Running pytest-cov along with pytest in VS Code - Stack Overflow](https://stackoverflow.com/questions/69443189/running-pytest-cov-along-with-pytest-in-vs-code#:~:text=Now%20you%20can%20run%20the%20tests%20with%20the), but it causes some known issues with debugger - [Debugger doesn't stop at breakpoints with pytest if pytest-cov is used · Issue #693 · microsoft/vscode-python (github.com)](https://github.com/microsoft/vscode-python/issues/693#issuecomment-926591721) forwhich there might be a workaround - [Testing Python in Visual Studio Code](https://code.visualstudio.com/docs/python/testing#_pytest-configuration-settings). Needs further testing. As agreed, I will close this PR as it is right now and we can have this as a possible improvement.

File: DEVELOPMENT.md (hunk 1 above)
Comment (jayachithra) [reply DEVELOPMENT.md:49]: Done!

**Review by arjendev — APPROVED** [review]

**Pull request merged**